labgrid/driver/sshdriver.py

```diff
@@ -72,19 +72,32 @@ def _start_own_master(self):
         env = os.environ.copy()
         if self.networkservice.password:
             env['SSHPASS'] = self.networkservice.password
-        self.process = subprocess.Popen(args, env=env)
+        self.process = subprocess.Popen(args, env=env,
+                                        stdout=subprocess.PIPE,
+                                        stderr=subprocess.PIPE,
+                                        stdin=subprocess.DEVNULL)
 
         try:
-            if self.process.wait(timeout=30) != 0:
+            subprocess_timeout = 30
+            return_value = self.process.wait(timeout=subprocess_timeout)
+            if return_value != 0:
+                stdout = self.process.stdout.readlines()
+                stderr = self.process.stderr.readlines()
                 raise ExecutionError(
-                    "failed to connect to {} with {} and {}".
-                    format(self.networkservice.address, args, self.process.wait())
+                    "Failed to connect to {} with {} and {}".
+                    format(self.networkservice.address, args, return_value),
+                    stdout=stdout,
+                    stderr=stderr
                 )
         except subprocess.TimeoutExpired:
+            stdout = self.process.stdout.readlines()
```

Comment:
This will block until the process either closes or writes a line. It defeats the purpose of the upper timeout. Must change this or remove it.

(the original motivation was to at least try to give some info on what the subprocess printed out before we killed it with the timeout from `self.process.wait`

Reply:
There seems to not be a quick fix for this. The [solution I've encountered](https://stackoverflow.com/a/4896288/1214214) basically suggest implementing an async read in a separated thread with a queue. 

Dropping the attempt to show std* data on a forced timeout.